### PR TITLE
Add configuration to disable unused license warnings

### PR DESCRIPTION
**Pull request by jonapoul**
I put this together because I had a similar use case to some of the people in https://github.com/cashapp/licensee/issues/134. Basically we have a list of allowed licenses in our organisation, so in our case it doesn't make sense to have a log full of warnings when there's no problem to flag up. Much simpler to configure the project with the licenses that we can use, then only get notified of any violations of that list.

Core of the change is a new method to the `LicenseeExtension`:
```kotlin
licensee {
  unusedAction(UnusedAction.IGNORE) // or WARN
}
```
where the default is `UnusedAction.WARN`. Most of the diffs are test fixtures, I haven't done a whole lot of Gradle plugin tests before so let me know whether these are decent.

### CODE REVIEW

File: src/main/kotlin/app/cash/licensee/pluginExtension.kt [hunk 1]
```diff
@@ -213,6 +235,12 @@ enum class ViolationAction {
   IGNORE,
 }
 
+@Suppress("unused") // Public API.
+enum class UnusedAction {
+  WARN,
```
Comment (JakeWharton) [inline src/main/kotlin/app/cash/licensee/pluginExtension.kt]: I think should probably be named `LOG` to match the violation action.

**Review by JakeWharton — APPROVED** [review]
Very nicely done! Just one small naming nit.

File: src/main/kotlin/app/cash/licensee/pluginExtension.kt (hunk 1 above)
Comment (jonapoul) [reply src/main/kotlin/app/cash/licensee/pluginExtension.kt]: Good eye - updated

**Comment by jonapoul** [comment]
Just tried running the build on my fork, seems to be setting up Java okay: https://github.com/jonapoul/licensee/actions/runs/7547641594
Not sure why it's failing here...

**Comment by JakeWharton** [comment]
Yeah I retried a few times already. I'll try again later.

**Pull request merged**